agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -124,25 +124,28 @@ private static Configuration loadConfigurationFile(Path agentJarPath) throws IOE
             return new Configuration();
         }
 
-        Path configPath;
-        boolean warnIfMissing;
         String configPathStr = getEnvVarOrProperty(APPLICATIONINSIGHTS_CONFIGURATION_FILE, "applicationinsights.configuration.file");
-        if (configPathStr == null) {
-            configPath = agentJarPath.resolveSibling("ApplicationInsights.json");
-            warnIfMissing = false;
-        } else {
-            configPath = agentJarPath.resolveSibling(configPathStr);
-            warnIfMissing = true;
+        if (configPathStr != null) {
+            Path configPath = agentJarPath.resolveSibling(configPathStr);
+            if (Files.exists(configPath)) {
+                return loadJsonConfigFile(configPath);
+            } else {
+                // fail fast any time configuration is invalid
+                throw new IllegalStateException("could not find requested configuration file: " + configPathStr);
+            }
         }
 
+        Path configPath = agentJarPath.resolveSibling("applicationinsights.json");
         if (Files.exists(configPath)) {
             return loadJsonConfigFile(configPath);
-        } else {
-            if (warnIfMissing) {
-                configurationMessages.add(new ConfigurationMessage("could not find configuration file: {}", configPathStr));
-            }
-            return new Configuration();
         }
+
+        if (Files.exists(agentJarPath.resolveSibling("ApplicationInsights.json"))) {
+            throw new IllegalStateException("found ApplicationInsights.json, but it should be lowercase: applicationinsights.json");
+        }
```

Comment:
@trask any particular reason to change json file name to lower case?

Reply:
oh yes, to be consistent with the other files in the same dir (and because it's a relic of us thinking Windows first 😄)

* applicationinsights-agent-3.0.0.jar
* applicationinsights.log
* applicationinsights.json